sync/fromDisk/questions.js

```diff
@@ -4,6 +4,36 @@ const sqldb = require('../../prairielib/lib/sql-db');
 const infofile = require('../infofile');
 const perf = require('../performance')('question');
 
+/**
+ * Based upon the passed question and a grading type,
+ *  returns a boolean if the question has the given grading method type
+ * @param {import('../course-db').Question} q
+ * @param {import('../course-db').GradingMethod} gradingMethodType
+ * @returns {boolean}
+ */
+function hasGradingMethod(q, gradingMethodType) {
+  const { gradingMethod, gradingMethods } = q;
+  if (gradingMethods) {
+    return Array.isArray(gradingMethods) && gradingMethods.indexOf(gradingMethodType) > -1;
+  }
+  return gradingMethod === gradingMethodType;
+}
+
+/**
+ * Returns true iff the given question is an internal grading question
+ * @param {import('../course-db').Question} q
+ * @returns {boolean}
+ */
+function hasInternalGrading(q) {
+  const { gradingMethod, gradingMethods } = q;
+  if (gradingMethods) {
+    return (
+      (Array.isArray(gradingMethods) && !gradingMethods.length) || hasGradingMethod(q, 'Internal')
```

Comment:
I don't 100% understand what's going on here - does this mean that an empty array of `gradingMethods` implies internal grading is enabled?

Reply:
Line 29: Null check for `gradingMethods`. In this case, fallback to `gradingMethod`.
Line 31: Yes, no `gradingMethods` implies Internal grading (just like no `gradingMethod` implies Internal (see line 34).